> Actually, wait a minute--@pablogsal, could you reproduce this on main or 3.13? I can't with either the test case or the reproducer provided in the issue. I don't see how this could possibly be platform-dependent.

Humm good point. In macOS I am seeing this:

```
Python 3.14.0a7+ (heads/main-dirty:3dfed230928, May  6 2025, 16:15:52) [Clang 17.0.0 (clang-1700.0.13.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> compile(source=b'Fal\xc5\xbfe', filename="<test>", mode="exec")
Traceback (most recent call last):
  File "<python-input-0>", line 1, in <module>
    compile(source=b'Fal\xc5\xbfe', filename="<test>", mode="exec")
    ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: identifier field can't represent 'False' constant
```

--- a/Lib/test/test_ast/test_ast.py
+++ b/Lib/test/test_ast/test_ast.py
@@ -822,10 +822,15 @@ class AST_Tests(unittest.TestCase):
                 compile(expr, "<test>", "eval")
 
     def test_constant_as_unicode_name(self):
-        for constant in b"Tru\xe1\xb5\x89", b"Fal\xc5\xbfe", b"N\xc2\xbane":
+        constants = [
+            ("True", b"Tru\xe1\xb5\x89"),
+            ("False", b"Fal\xc5\xbfe"),
+            ("None", b"N\xc2\xbane"),
+        ]
+        for constant in constants:
             with self.assertRaisesRegex(ValueError,
-                "identifier must not be None, True or False after Unicode normalization \\(NKFC\\)"):
-                ast.parse(constant, mode="eval")
+                f"identifier field can't represent '{constant[0]}' constant"):
+                ast.parse(constant[1], mode="eval")
 
     def test_precedence_enum(self):
         class _Precedence(enum.IntEnum):

--- a/Parser/pegen.c
+++ b/Parser/pegen.c
@@ -549,11 +549,13 @@ _PyPegen_new_identifier(Parser *p, const char *n)
         }
         id = id2;
     }
-    if (_PyUnicode_EqualToASCIIString(id, "None") ||
-        _PyUnicode_EqualToASCIIString(id, "True") ||
-        _PyUnicode_EqualToASCIIString(id, "False"))
+    if (_PyUnicode_EqualToASCIIString(id, "None")
+        || _PyUnicode_EqualToASCIIString(id, "True")
+        || _PyUnicode_EqualToASCIIString(id, "False"))
     {
-        PyErr_SetString(PyExc_ValueError, "identifier must not be None, True or False after Unicode normalization (NKFC)");
+        PyErr_SetString(PyExc_ValueError,
+                        "identifier must not be None, True or False "
+                        "after Unicode normalization (NKFC)");
         Py_DECREF(id);
         goto error;
     }

--- a/Parser/pegen.c
+++ b/Parser/pegen.c
@@ -549,15 +549,20 @@ _PyPegen_new_identifier(Parser *p, const char *n)
         }
         id = id2;
     }
-    if (_PyUnicode_EqualToASCIIString(id, "None")
-        || _PyUnicode_EqualToASCIIString(id, "True")
-        || _PyUnicode_EqualToASCIIString(id, "False"))
-    {
-        PyErr_SetString(PyExc_ValueError,
-                        "identifier must not be None, True or False "
-                        "after Unicode normalization (NKFC)");
-        Py_DECREF(id);
-        goto error;
+    static const char * const forbidden[] = {
+        "None",
+        "True",
+        "False",
+        NULL
+    };
+    for (int i = 0; forbidden[i] != NULL; i++) {
+        if (_PyUnicode_EqualToASCIIString(id, forbidden[i])) {
+            PyErr_Format(PyExc_ValueError,
+                         "identifier field can't represent '%s' constant",
+                         forbidden[i]);
+            Py_DECREF(id);
+            goto error;
+        }
     }
     PyInterpreterState *interp = _PyInterpreterState_GET();
     _PyUnicode_InternImmortal(interp, &id);

